server.js

```diff
@@ -7,14 +7,70 @@ app.use(express.json());
 app.use(cors());
 
 //Use this array as your (in-memory) data store.
-const bookings = require("./bookings.json");
+let bookings = require("./bookings.json");
+const { request, response } = require("express");
+const { json } = require("express/lib/response");
+// const { request, response } = require("express");
 
 app.get("/", function (request, response) {
   response.send("Hotel booking server.  Ask for /bookings, etc.");
 });
 
 // TODO add your routes and helper functions here
 
+//missing properties --> don't add
+//extra properties ---> add but ignore unfamiliar ones
+
+//empty object ---> don't add
+const isObjEmpty = (bookingObj) => {
+  return Object.keys(bookingObj).length === 0;
+};
+
+//correct object ---> add
+// const objHasCorrectProperties = (bookingObj) => {
+//   if ("roomId" in bookingObj 
+//       && "title" in bookingObj
+//       && "roomId" in bookingObj
+//       && "firstName" in bookingObj
+//       && "surname" in bookingObj 
+//       && "email" in bookingObj
+//       && "checkInDate" in bookingObj 
+//       && "checkOutDate" in bookingObj
+//       ){
+//         bookings.push();
+//         return response.send("Added booking");
+//       } else {
+//         return response.sendStatus(404);
+//       }; 
+// };
+
+//Read one booking, specified by an ID
+app.get("/bookings/:id", (request, response) => {
+  response.send(bookings.find((booking) => booking.id === Number(request.params.id)));
+});
+
+//Delete a booking, specified by an ID
+app.delete("/bookings/:id", (request, response) => {
+  bookings = bookings.filter(booking => booking.id != request.params.id);
```

Comment:
remember to use strict equals/not equals, i.e. always use `!==` instead of `!=`, which might mean you have to write `booking.id !== Number(request.params.id)`